packages/core/src/readers/SimpleDirectoryReader.edge.ts

```diff
@@ -46,19 +47,53 @@ export class SimpleDirectoryReader implements BaseReader {
     const {
       directoryPath,
       fs = defaultFS,
-      defaultReader = new TextFileReader(),
+      defaultReader,
       fileExtToReader,
+      numWorkers = 1,
     } = params;
 
+    // Check if LlamaParseReader is used as the defaultReader and if so checks if numWorkers is in the valid range
+    if (defaultReader instanceof LlamaParseReader && numWorkers > 9) {
+      throw new Error("Currently, LlamaParse supports a maximum of 9 workers.");
+    }
+
     // Observer can decide to skip the directory
     if (
       !this.doObserverCheck("directory", directoryPath, ReaderStatus.STARTED)
     ) {
       return [];
     }
 
-    const docs: Document[] = [];
+    const filePathQueue: string[] = [];
+
     for await (const filePath of walk(fs, directoryPath)) {
+      filePathQueue.push(filePath);
+    }
+
+    const workerPromises = Array.from({ length: numWorkers }, () =>
```

Comment:
is this handling queue.length > numWorkers? maybe using something like p-limit helps?

Reply:
Yes it is, I thought about using p-limit, I just always try to avoid adding packages if possible. But will check it out and test around. Thanks for the review!